examples/subscribe_invoices.rs

```diff
@@ -0,0 +1,30 @@
+// This program accepts three arguments: address, cert file, macaroon file
+// The address must start with `https://`!
+
+#[tokio::main]
+async fn main() {
+    let mut args = std::env::args_os();
+    args.next().expect("not even zeroth arg given");
+    let address = args.next().expect("missing arguments: address, cert file, macaroon file");
+    let cert_file = args.next().expect("missing arguments: cert file, macaroon file");
+    let macaroon_file = args.next().expect("missing argument: macaroon file");
+    let address = address.into_string().expect("address is not UTF-8");
+
+    // Connecting to LND requires only address, cert file, and macaroon file
+    let mut client = tonic_lnd::connect(address, cert_file, macaroon_file)
+        .await
+        .expect("failed to connect");
+
+    let mut invoice_stream = client
+        .subscribe_invoices(tonic_lnd::rpc::InvoiceSubscription { add_index: 0, settle_index: 0 })
+        .await
+        .expect("Failed to call subscribe_invoices")
+        .into_inner();
+
+    while let Some(invoice) = invoice_stream.message().await.expect("Failed to receive invoices") {
+        // If this invoice was paid we can do something with it
+        if invoice.settle_date > 0 {
```

Comment:
Isn't checking `settled` field better? (I may be misremembereing something)

Reply:
Yeah I used that at first but turns out settled is deprecated

Reply:
If you don't like use `settle_date` we can use `state == "Open"`

Reply:
I think is a good idea, I fixed it to compare the state instead